### ISSUE TITLE
Optimize Move()

### ISSUE THREAD

**Issue by JasonBock** [JasonBock/IronBefunge#3]
Right now the Move() method in ExecutionContext is horrible. It goes through the FungeSpace one cell at a time, so if the next instruction is 1000 points away ... well, that sucks. Need to come up with a way to not only find out the next valid cell in the current direction, but also cache those values as cell references to remove the need for more lookups.

**Comment by JasonBock** [comment]
As I found out tonight, I had to create `maxX` and `maxY` values in `ExecutionContext`. Using `MaxValue` for `int` was really dumb :). However, I can still optimize finding the next `CurrentPosition`. Maybe instead of just incrementing or decrementing the value, I set it to the `x` or `y` value above or below the current position using the `Max` or `Min` LINQ functions. This would also eliminate the `null` check from `Find` because you'd always be guaranteed to have the next instruction (well, maybe, unless other places are calling `Find` - would have to verify that).
